### PR TITLE
Hr diagram

### PR DESCRIPTION
**Pull request by guynir42**
Scans each source's annotations for info from GAIA (this can be replaced with other catalogs). 
If an annotation has a key named GAIA and it has color/magnitude keys in it, then the source gets an object with 
{origin: origin, abs_mag: abs_mag, color: color}. 
This is read by the Vega plot and displays all origins available for that source. 
By default each source is scanned this way but I don't think this is very slow, and it gives an indication for the user if this source is stellar or transient by checking if "color_magnitude" is empty. 

HR_diagram_source_table

### CODE REVIEW

**Comment by guynir42** [comment]
Also I need to add tests and integrate this component into the source page (right now it is only on the source tables).

File: skyportal/handlers/api/color_mag.py
Comment (profjsb) [inline skyportal/handlers/api/color_mag.py:34]:
```suggestion
```

File: skyportal/handlers/api/color_mag.py
Comment (profjsb) [inline skyportal/handlers/api/color_mag.py:45]:
```suggestion
        for key, xmatch in an.data.items():  
        # go over all items in the data (e.g., different catalog matches)
```

File: skyportal/handlers/api/color_mag.py
Comment (profjsb) [inline skyportal/handlers/api/color_mag.py:90]:
```suggestion
                if  absorption_key is not None:  
                # only check this if given an absorption term
```

File: static/js/components/SourceTable.jsx
Comment (profjsb) [inline static/js/components/SourceTable.jsx:331]: Should the width/height be set by the context (ie. a percentage of the screen, rather than size; mobile vs desktop)?

File: static/js/components/VegaHR.jsx
Comment (profjsb) [inline static/js/components/VegaHR.jsx:9]: Should padding be a fraction of width/height?

File: static/js/components/VegaHR.jsx
Comment (profjsb) [inline static/js/components/VegaHR.jsx:11]:
```suggestion
    // draw the Gaia data for main sequence etc.
```

File: static/js/components/VegaHR.jsx
Comment (profjsb) [inline static/js/components/VegaHR.jsx:38]: Unclear from this labeling if this is absolute G or not.

**Review by profjsb — COMMENTED** [review]
Looks good. Some small formatting changes here.

File: skyportal/handlers/api/source.py
Comment (acrellin) [inline skyportal/handlers/api/source.py:657]: Will this be widely relevant for most API users? If not, we should make this optional via a query param that defaults to false

**Review by acrellin — CHANGES_REQUESTED** [review]
In addition to below comments, the new API endpoint needs a test

File: static/js/ducks/source.js
Comment (acrellin) [inline static/js/ducks/source.js:107]: Why keep this if it's never used? Same for action type above

**Comment by acrellin** [comment]
@guynir42 looking good. I'd suggest removing the API endpoint/handler stuff that's not used. The function should be tested, still, however.

File: static/js/components/SourceTable.jsx
Comment (guynir42) [reply static/js/components/SourceTable.jsx:331]: I'm not sure. I think the component should not be deciding for itself, rather let the surrounding page choose the size for it.

File: skyportal/handlers/api/color_mag.py
Comment (guynir42) [reply skyportal/handlers/api/color_mag.py:48]: Yes, I need to break bad habits...

File: skyportal/handlers/api/source.py
Comment (guynir42) [reply skyportal/handlers/api/source.py:657]: I thought about this, but then you will want to have it true for all the source tables. The idea is that this is how you know if the source is transient or not.

File: static/js/ducks/source.js
Comment (guynir42) [reply static/js/ducks/source.js:107]: Yes you are right, I forgot about that.

File: skyportal/handlers/api/color_mag.py
Comment (guynir42) [reply skyportal/handlers/api/color_mag.py:215]: I'm not sure if I got everything but does this look ok? 
`responses:
      200:
        content:
          application/json:
            schema:
              allOf:
                  - $ref: '#/components/schemas/Success'
                  - type: array
                    items:
                      type: object
                      properties:
                          origin:
                            type: string
                          color:
                            type: float
                          abs_mag:
                            type: float

      400:
        content:
        application/json:
          schema: Error`

**Comment by guynir42** [comment]
I have a weird issue. When I add the vega plot to the source page, it has some problem reading the background JSON file:

Screenshot from 2021-03-04 16-19-53

Any ideas what is going on? @acrellin @dmitryduev

File: static/HR_density.json
Comment (dmitryduev) [inline static/HR_density.json:1]: This is really nice!

**Review by dmitryduev — COMMENTED** [review]
I really like how it's coming together @guynir42! As discussed, please add some demo data (with some basic tests) + place the diagram on the right-hand side in its own accordion on the Source page -> then should be ready to go in.

File: static/HR_density.json
Comment (guynir42) [reply static/HR_density.json:1]: Thanks! also wanted to ask: is there a trick to checking the xpath for a vega plot? I added `data-testid` but I don't see it anywhere on the inspect element and it doesn't find it when running the test...

**Comment by dmitryduev** [comment]
@guynir42 would you mind posting the latest screen shots here?

File: skyportal/handlers/api/source.py
Comment (acrellin) [inline skyportal/handlers/api/source.py:280]: Description needs updating

**Review by acrellin — APPROVED** [review]
Looks good, thanks @guynir42 , just the one comment below re: docstring update

**Review by dmitryduev — APPROVED** [review]
Excellent work @guynir42! LGTM, this should go in if other reviewer are happy. 

I posted a related issue on F https://github.com/fritz-marshal/fritz/issues/197, would you please document this wonderful new feature?

**Comment by guynir42** [comment]
> Excellent work @guynir42! LGTM, this should go in if other reviewer are happy.
> 
> I posted a related issue on F [fritz-marshal/fritz#197](https://github.com/fritz-marshal/fritz/issues/197), would you please document this wonderful new feature?

Thanks!

I will try to get it done today/tomorrow. 

@acrellin  can we get an approval + merge on this one?

**Comment by guynir42** [comment]
Hey @acrellin could you merge this one?

**Comment by stefanv** [comment]
@guynir42 I'm going to merge now.

What script did you use to generate the HR json data file?  Can this be added in a next PR?

**Comment by guynir42** [comment]
> @guynir42 I'm going to merge now.
> 
> What script did you use to generate the HR json data file? Can this be added in a next PR?

I got it from Jan van Roestel.

**Comment by stefanv** [comment]
Can you find out where he got it / how he made it?

**Pull request merged**
